### ISSUE TITLE
[html] Reported range is incorrect for UseAltAttributeForImages rule

### ISSUE THREAD

**Issue by SCWells72** [pmd/pmd#3951]
**Affects PMD Version:** 

6.45.0

**Description:**

I'm integrating the new HTML language support into my IDE plugin (Illuminated Cloud 2) and noticed that the reported violation range for `UseAltAttributeForImages` is incorrect. For example, the violation reported in the following document:

```html
<html>
    <img src="house.jpg">
</html>
```

has `beginLine=2, beginColumn=5, endLine=3, endColumn=6` which spans the `img` tag but also into the next line's closing `html` tag.

I know this new language support is beta but figured it doesn't hurt to report this incorrect violation data.

**Comment by adangel** [comment]
Thanks for reporting. Yes, as with all DOM related parsers, the original locations are not provided, so we try to guess them manually. It's probably the special case of a tag, which doesn't have a closing tag, like `<img>`. Same problem probably appears with `<meta>`.

**Comment by UncleOwen** [comment]
With 7.16.0 I get
```xml
<violation beginline="2" endline="2" begincolumn="5" endcolumn="26" rule="UseAltAttributeForImages" ruleset="Best Practices" externalInfoUrl="${pmd.website.baseurl}/pmd_rules_html_bestpractices.html#usealtattributeforimages" priority="3">
The image is missing an alternate text.
</violation>
```
Seems to be fixed.

**Comment by adangel** [comment]
Indeed, has been fixed by #5328 for PMD 7.8.0.
